microprofile/WFLY-10712_microprofile_metrics.asciidoc

```diff
@@ -0,0 +1,230 @@
+  `401 NOT AUTHORIZED` response):
+
+----
+$ curl -v http://127.0.0.1:9990/metrics/
+...
+< HTTP/1.1 401 Unauthorized
+----
+
+In that case, WildFly MUST have a management user and the HTTP client MUST pass its credential to the Metrics HTTP endpoint:
+
+----
+$ curl -v --digest -u admin:adminpwd http://127.0.0.1:9990/metrics
+< HTTP/1.1 200 OK
+...
+# HELP base:cpu_system_load_average Displays the system load average for the last minute. The system load average is the sum of the number of runnable entities queued to the available processors and the number of runnable entities running on the available processors averaged over a period of time. The way in which the load average is calculated is operating system specific but is typically a damped time-dependent average. If the load average is not available, a negative value is displayed. This attribute is designed to provide a hint about the system load and may be queried frequently. The load average may be unavailable on some platform where it is expensive to implement this method.
+# TYPE base:cpu_system_load_average gauge
+base:cpu_system_load_average 1.9658203125
+...
+----
+
+If security is disabled in the subsystem configuration, the HTTP client does not require authentication (and it does not require a WildFly management user either).
+
+Note that the _standalone profiles will explicitly disable authentication_ as the HTTP management interface is bound to `127.0.0.1`.
+
+==== Exposed Metrics
+
+===== Base Metrics
+
+By default, the subsystem will expose all required base metrics specified in the chapter 4 of the MicroProfile Metrics specification.
+All these base metrics are gathered from the JVM MBeans and uses smallrye-metrics JmxRegistrar to bridge from JMX to the MicroProfile Metrics API.
+The required base metrics are explicitly specified in a property file (named `base-metrics.properties`) using a set of properties for each metric:
+
+----
+<metric name>.displayName: <Human-readable name of the metric>
+<metric name>.type: <Type of metric enumerated in org.eclipse.microprofile.metrics.MetricType (e.g counter, gauge)>
+<metric name>.unit: <Unit of the metric, can be none, listed in org.eclipse.microprofile.metrics.MetricUnits or other units>
+<metric name.description: <Human-readable description of the metric>
+<metric name>.mbean: <ObjectName of the MBean and attribute>
+----
+
+For example, the properties to expose the _Total Loaded Class Count_ of the JVM are:
+
+----
+classloader.totalLoadedClass.count.displayName: Total Loaded Class Count
+classloader.totalLoadedClass.count.type: counter
+classloader.totalLoadedClass.count.unit: none
+classloader.totalLoadedClass.count.description: Displays the total number of classes that have been loaded since the Java virtual machine has started execution.
+classloader.totalLoadedClass.count.mbean: java.lang:type=ClassLoading/TotalLoadedClassCount
```

Comment:
uff, now I'm confused when `:` and `.` should be used as separators
e.g. this vs. base:cpu_system_load_average